test/Model.spec.js

```diff
@@ -347,101 +358,137 @@ describe("Model", function() {
-                },
-                function(error) {
-                    onErrorCalled++;
-                },
-                function() {
-                    onCompletedCalled++;
-                });
+        const subscription = rxjs.Observable.from(model.get("list[0,1].name")).subscribe(
+            value => {
+                onNextCalled++;
+            },
+            error => {
+                onErrorCalled++;
+            },
+            () => {
+                onCompletedCalled++;
+            }
+        );
 
         subscription.unsubscribe();
 
-        if (dataSourceGetCalled === 1 && !onNextCalled && unusubscribeCalled === 1 && !onErrorCalled && !onCompletedCalled) {
-            done()
-        }
-        else {
+        if (
+            dataSourceGetCalled === 1 &&
+            !onNextCalled &&
+            unusubscribeCalled === 1 &&
+            !onErrorCalled &&
+            !onCompletedCalled
+        ) {
+            done();
+        } else {
             done(new Error("DataSource unsubscribe not called."));
         }
     });
 
-    it('setMaxSize to a lower value forces a collect', function(done) {
-        var model = new Model({
+    it("setMaxSize to a lower value forces a collect", () => {
+        const model = new Model({
             cache: {
                 list: {
                     0: { name: "test" }
                 }
             }
         });
-        var cache = model._root.cache;
-        expect(cache['$size']).toBeGreaterThan(0);
+        const cache = model._root.cache;
+        expect(cache.$size).toBeGreaterThan(0);
         model._setMaxSize(0);
-        expect(cache['$size']).toBe(0);
-        done();
+        expect(cache.$size).toBe(0);
     });
 
     // https://github.com/Netflix/falcor/issues/915
-    it('maxRetries option is carried over to cloned Model instance', function(done) {
-        var model = new Model({
+    it("maxRetries option is carried over to cloned Model instance", () => {
+        const model = new Model({
             maxRetries: 10
         });
         expect(model._maxRetries).toBe(10);
-        var batchingModel = model.batch(100);
+        const batchingModel = model.batch(100);
         expect(batchingModel._maxRetries).toBe(10);
-        done();
     });
 
-    it('cloned instance should retain custom type', function() {
+    it("cloned instance should retain custom type", () => {
         function MyModel() {
             Model.call(this);
         }
         MyModel.prototype = new Model();
         MyModel.prototype.constructor = MyModel;
-        var model = new MyModel();
+        const model = new MyModel();
         expect(model).toBeInstanceOf(MyModel);
-        var clonedModel = model.batch(100);
+        const clonedModel = model.batch(100);
         expect(clonedModel).toBeInstanceOf(MyModel);
     });
 
-    describe('JSON-Graph Specification', function() {
-        require('./get-core');
+    describe("algorithm options", () => {
```

Comment:
instead of testing all algorithm options together, i'd prefer them tested separately, just because we may have more algorithm-related options later which may not be orthogonal to each other in the way these two are.

i propose the following structure for the test hierarchy here:
```
describe('Model...
    describe('disable path collapse algorithm'...
    describe('disable request de-duplication'...
```